### PR TITLE
workflows: add scheduled job to sync `openshift/os` submodule

### LINKED ISSUE #709: Set up Dependabot to bump fedora-coreos-config submodule regularly
We should automate these bumps rather than rely on humans doing it.

### PR DESCRIPTION
**Pull request by bgilbert**
We've been hand-maintaining the `fedora-coreos-config` submodule in `openshift/os`, but updating it is usually completely mechanical.  Add a workflow that maintains an `openshift/os` PR with the latest changes, updating it daily or on demand.  We can merge the PR whenever we'd like, and the job will open a new one once there are additional changes.

Omit CoreOS Bot changes from the log summary, since those involve lockfiles and overrides that don't apply to RHCOS.  Likewise, if the only changes are from CoreOS Bot, skip opening a PR.

When the workflow is manually invoked, allow overriding the PR branch.  This can be used to create a one-off PR that won't be maintained by the bot afterward, for cases where a sync needs complementary changes elsewhere in `openshift/os`.

Ideally the job would run inside the `openshift/os` repo, but GitHub Actions isn't available there.

Fixes https://github.com/openshift/os/issues/709.

### CODE REVIEW

**Review by travier — DISMISSED** [review]

File: .github/workflows/openshift-os.yml [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+          git submodule init
+          git submodule update
+
+          cd fedora-coreos-config
+          # Omit CoreOS Bot commits from the log message, since they generally
+          # only affect FCOS
+          git log "HEAD..testing-devel" --perl-regexp \
+              --author='^((?!CoreOS Bot <coreosbot@fedoraproject.org>).*)$' |
+              git shortlog > $RUNNER_TEMP/shortlog
```
Comment (jlebon) [inline .github/workflows/openshift-os.yml]: Minor: why not just invoke `git shortlog` directly? The same switches are supported, and ranges too.

**Review by jlebon — DISMISSED** [review]
Nice! One minor comment, but LGTM as is.

File: .github/workflows/openshift-os.yml (hunk 1 above)
Comment (bgilbert) [reply .github/workflows/openshift-os.yml]: ...ah.  I learned `git shortlog` in the stone age, when it was only a postprocessor.  Updated.

**Review by jlebon — APPROVED** [review]

**Comment by dustymabe** [comment]
> Ideally the job would run inside the `openshift/os` repo, but GitHub Actions isn't available there.

Indeed. I think in the future if our abilities change there we should definitely move this.

**Review by dustymabe — APPROVED** [review]
I have no objections here. Thank you for working on it!

**Comment by bgilbert** [comment]
CI doesn't test this.  Merging.

**Pull request merged**